ui/app/components/ui/copyButton.js

```diff
@@ -1,66 +1,66 @@
-const Component = require('react').Component
-const PropTypes = require('prop-types')
-const h = require('react-hyperscript')
-const inherits = require('util').inherits
-const copyToClipboard = require('copy-to-clipboard')
-const connect = require('react-redux').connect
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
 
+const copyToClipboard = require('copy-to-clipboard')
 const Tooltip = require('./tooltip')
 
-CopyButton.contextTypes = {
-  t: PropTypes.func,
-}
-
-module.exports = connect()(CopyButton)
-
-
-inherits(CopyButton, Component)
-function CopyButton () {
-  Component.call(this)
-}
+class CopyButton extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
 
-// As parameters, accepts:
-// "value", which is the value to copy (mandatory)
-// "title", which is the text to show on hover (optional, defaults to 'Copy')
-CopyButton.prototype.render = function () {
-  const props = this.props
-  const state = this.state || {}
+  static defaultProps = {
+    title: null,
+  }
 
-  const value = props.value
-  const copied = state.copied
+  static propTypes = {
+    value: PropTypes.any.isRequired,
```

Comment:
```suggestion
    value: PropTypes.string.isRequired,
```
The `copy-to-clipboard` library accepts only `string` input.

Reply:
... after further investigation, the value passed to this component for the `value` prop is always `undefined` :unamused: . So this is broken.

Reply:
Wait, uhhh hmm?

Reply:
Here's the only place that prop is passed in: https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/components/app/shift-list-item.js#L77

`props.depositAddress` doesn't exist. It's always undefined.

Reply:
Thanks, I hate it